Lib/fontTools/ttLib/tables/_g_v_a_r.py

```diff
@@ -116,32 +124,26 @@ def decompile(self, data, ttFont):
         sstruct.unpack(GVAR_HEADER_FORMAT, data[0:GVAR_HEADER_SIZE], self)
         assert len(glyphs) == self.glyphCount
         assert len(axisTags) == self.axisCount
-        offsets = self.decompileOffsets_(
-            data[GVAR_HEADER_SIZE:],
-            tableFormat=(self.flags & 1),
-            glyphCount=self.glyphCount,
-        )
         sharedCoords = tv.decompileSharedTuples(
             axisTags, self.sharedTupleCount, data, self.offsetToSharedTuples
         )
         variations = {}
         offsetToData = self.offsetToGlyphVariationData
         glyf = ttFont["glyf"]
 
-        def decompileVarGlyph(glyphName, gid):
-            gvarData = data[
-                offsetToData + offsets[gid] : offsetToData + offsets[gid + 1]
-            ]
-            if not gvarData:
-                return []
-            glyph = glyf[glyphName]
-            numPointsInGlyph = self.getNumPoints_(glyph)
-            return decompileGlyph_(numPointsInGlyph, sharedCoords, axisTags, gvarData)
+        l = LazyDict(
+            {glyphs[gid]: _decompileVarGlyph for gid in range(self.glyphCount)}
+        )
+        l.reverseGlyphMap = ttFont.getReverseGlyphMap()
+        l._glyf = glyf
+        l._gvar = self
+        l.gvarData = data
+        l.offsetToData = offsetToData
+        l.sharedCoords = sharedCoords
+        l.axisTags = axisTags
+        l.tableFormat = self.flags & 1
```

Comment:
why turn `_decompileVarGlyph` into a kind-of instance method (that takes self but is defined as standalone function) and not keep it as a closure as it was before? you wouldn't need to set all these attribute on the LazyDict instance, they'd be captured from the outer namespace.